index.js

```diff
@@ -524,7 +541,15 @@ function flameGraph (opts) {
   // Wait for 500 ms before showing the tooltip.
   var tooltipFocusNode = null
   var tooltipFocusTimeout = null
+  var hoveringIn = false
   function showTooltip (node) {
+
+    //let's dispatch the hover event with no delay
+    const pointerCoords = {x: d3.event.offsetX, y: d3.event.offsetY}
+    dispatch.call('hoverin', null, {...node, rect: getNodeRect(node), pointerCoords})
```

Comment:
A few things about the arguments to this event:

- We can emit only `node.data` here, rather than the entire `node`. Only the `data` property is useful to outside consumers, the rest is d3-fg internal state.

- Merging the `rect` property into the node here means the `node` object has a different shape between the `zoom` and `hoverin` events, which could potentially be confusing. Maybe we could pass the `rect` object as a separate argument instead, so the callback would do `dispatch.on('hoverin', (node, rect) => {})`?

- Passing `pointerCoords` may not be necessary since Flame can also access `d3.event.offsetX/Y` on its own. I don't mind having it if you find it clearer this way though. But then it should be a third argument, I think.

Reply:
Good points, I left `pointersCoords` even if not strictly needed as it could be even clearer this way